ta/avb/entry.c

```diff
@@ -212,9 +266,92 @@ static TEE_Result write_lock_state(uint32_t pt,
 	if (count == sizeof(lock_state) && lock_state == wlock_state)
 		goto out;
 
-	res = create_state(wlock_state, &h);
+	res = create_rb_state(wlock_state, &h);
+out:
+	TEE_CloseObject(h);
+	return res;
+}
+
+static TEE_Result write_persist_value(uint32_t pt,
+				      TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Result res;
+	TEE_ObjectHandle h;
+	char *name_buf;
+	uint32_t name_buf_sz;
+	char *value;
+	uint32_t value_sz;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	name_buf = params[0].memref.buffer;
+	name_buf_sz = params[0].memref.size;
+
+	value = params[1].memref.buffer;
+	value_sz = params[1].memref.size;
+
+	res = open_named_value(name_buf, name_buf_sz, value,
+			      value_sz, true, &h);
+	if (res)
+		return res;
+
+	res = TEE_SeekObjectData(h, 0, TEE_DATA_SEEK_SET);
+	if (res)
+		goto out;
+
+	res = TEE_WriteObjectData(h, value, value_sz);
+out:
+	TEE_CloseObject(h);
+
+	return res;
+}
+
+static TEE_Result read_persist_value(uint32_t pt,
+				      TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INOUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Result res;
+	TEE_ObjectHandle h;
+	char *value;
+	char *name_buf;
+	uint32_t name_buf_sz;
+	uint32_t value_sz;
+	uint32_t count;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	name_buf = params[0].memref.buffer;
+	name_buf_sz = params[0].memref.size;
+
+	value = params[1].memref.buffer;
+	value_sz = params[1].memref.size;
+
+	res = open_named_value(name_buf, name_buf_sz, value,
```

Comment:
It's a bit confusing that `value` and `value_sz` are supplied here as they will never be used.

Reply:
`347: res =  TEE_ReadObjectData(h, value, value_sz, &count);`

Reply:
Yes, but not by the function `open_named_value()`

Reply:
ah, sorry, you meant in `open_named_value()`

Reply:
fixed